src/lib/icu/meson.build

```diff
@@ -12,17 +12,23 @@ if is_windows
   icu_sources += 'Win32.cxx'
 endif
 
+iconv_dep = []
 if icu_dep.found()
   icu_sources += [
     'Util.cxx',
     'Init.cxx',
   ]
 elif not get_option('iconv').disabled()
-  have_iconv = compiler.has_function('iconv', prefix : '#include <iconv.h>')
-  conf.set('HAVE_ICONV', have_iconv)
+  # an installed iconv library will make the builtin iconv() unavailable,
+  # so search for the library first and pass it as (possible) dependency
+  iconv_dep = compiler.find_library('libiconv', required: false)
+  have_iconv = compiler.has_function('iconv', 
```

Comment:
This may actually do odd things (probably just over-linking) on systems where the installed iconv library is split into dev / lib packages, and only the lib package is installed. The builtin header declarations would be used, but you check them via linking to the external library, and then add the external library as an mpd dependency.

There are also *really* fun edge cases alluded to in https://github.com/mesonbuild/meson/commit/1ac6fc0595f992621cbb79b7971ba07db1fa8e62 whereby trying to detect has_function('iconv') will misfire on MinGW because idk, __has_builtin returns true over there. This is why I implemented a custom snippet for that check.

Reply:
The commit you linked to is libintl; you probably mean mesonbuild/meson@1e19757

Using has_function() is indeed buggy, it leads to false positives on some systems, because, with with no `__builtin` but an installed library, the test code does nothing and just returns 0:

```
#ifdef __has_builtin`
  #if !__has_builtin({__builtin_}{func})
    #error "{__builtin_}{func} not found"
  #endif
#elif ! defined({func})
  {__builtin_}{func};
#endif
return 0;
```

This is why I check for the library first; it also ensures that the library is linked. Linking against libiconv is currently missing. 

The custom iconv dependency() clause seems to be a 0.60 feature, I do not know if MPD is willing to upgrade to a newer version. With meson < 0.60.0 the dependency() clause will have the same issues as the code currently used by MPD ...

Reply:
Yeah, sorry for the mis-link. :)